components/controls/vehicles/VehicleRow.tsx

```diff
@@ -1,44 +1,56 @@
+import clsx from 'clsx'
 import React from 'react'
+import { View } from 'react-native'
 
 import FlexRow from '@/components/shared/FlexRow'
+import Icon from '@/components/shared/Icon'
 import IconButton from '@/components/shared/IconButton'
 import Panel from '@/components/shared/Panel'
 import Typography from '@/components/shared/Typography'
 import { useTranslation } from '@/hooks/useTranslation'
 import { Vehicle } from '@/hooks/useVehiclesStorage'
 
+type AdditionalProps =
+  | {
+      onContextMenuPress?: (licencePlate: string) => void
+      selected?: never
+      showControlChevron?: never
+    }
+  | {
+      onContextMenuPress?: never
+      selected?: boolean
+      showControlChevron?: never
+    }
+  | {
+      onContextMenuPress?: never
+      selected?: never
+      showControlChevron?: boolean
+    }
```

Comment:
this likely deserves extra comment - for example, I don't know what this achieves - is it that only one of the props can be passed at a time ?

Reply:
Okay, I'll add comment :) And yes, exactly.